tests/manifests/test_serviceaccounts.py

```diff
@@ -123,6 +125,6 @@ async def test_does_not_create_serviceaccounts_if_configured_not_to(values, make
             else:
                 covered_serviceaccount_names.add(serviceaccount_name)
 
-        assert serviceaccount_names == covered_serviceaccount_names, (
+        assert serviceaccount_names.intersection(covered_serviceaccount_names) == set(), (
```

Comment:
Because disabling service account creation does not disable setting the default service account name. This caused issues where the service account name was present but the service account properly absent, and the test failed.

```

E           AssertionError: Deployment/pytest-iwjypsiyvckpknjzhufqth-synapse-fed-reader (for synapse-federation-reader) created ServiceAccounts that it shouldn't have
E           assert set() == {'pytest-iwjy...fqth-synapse'}
E             
E             Extra items in the right set:
E             'pytest-iwjypsiyvckpknjzhufqth-synapse'
E             
E             Full diff:
E             + set()
E             - {
E             -     'pytest-iwjypsiyvckpknjzhufqth-synapse',
E             - }

```